### PR TITLE
fix bug: invalid VP increase via extendStakingDuration

### PR DESCRIPTION
**Pull request by tjcloa**

### CODE REVIEW

File: tests/staking/ExtendedStakingTest.js [hunk 1]
```diff
@@ -608,6 +608,34 @@ contract("Staking", (accounts) => {
             await expectRevert(staking.extendStakingDuration(lockedTS, newTime), "S04"); // S04 : cannot reduce staking duration
         });
 
+        it("Is protected from getting unlimited voting power by using corner case values in extendStakingDuration function", async () => {
+            let amount = "1337";
```
Comment (cwsnt) [inline tests/staking/ExtendedStakingTest.js:612]: Any reason why using this specific number?

Comment (tjcloa) [reply tests/staking/ExtendedStakingTest.js]: nah  
just copypasted from the vulnerability report  
changed for clarity though  
[commit](https://github.com/DistributedCollective/Sovryn-smart-contracts/commit/1c02fa2a1b5af97f735a0a52d41ee9c1be8abcaf)

**Review by ororopickpocket — COMMENTED** [review]
In general this should work.
I think we should consider to also change the order of setting the delegates from

```
        address delegateFrom = delegates[msg.sender][previousLock];
        address delegateTo = delegates[msg.sender][until];
        if (delegateTo == address(0)) {
            delegateTo = delegateFrom;
            delegates[msg.sender][until] = delegateFrom;
        }
        delegates[msg.sender][previousLock] = address(0);
```

to 

```
        address delegateFrom = delegates[msg.sender][previousLock];
        delegates[msg.sender][previousLock] = address(0);
        address delegateTo = delegates[msg.sender][until];
        if (delegateTo == address(0)) {
            delegateTo = delegateFrom;
            delegates[msg.sender][until] = delegateFrom;
        }  
```

What do you think, @tjcloa ?  

======================
@tjcloa agreed - adds one more level of protection

**Review by cwsnt — APPROVED** [review]
